elastic/elastic-agent
# Include upgrade details in output of `elastic-agent status` during ongoing upgrade

## What does this PR do?

This PR enhances the output of `elastic-agent status` to include upgrade details when Agent is being upgraded.

## Why is it important?

To gain more visibility into the Agent upgrade process.

## Checklist


## How to test this PR locally

1. Build `elastic-agent` from this PR's branch and install it.
2. Run `elastic-agent status --output json`. Verify that NO upgrade details are shown.
3. In a new window, upgrade Elastic Agent to a version that hasn't been released yet.  This will ensure that the upgrade starts but remains stalled.
   ```
   sudo elastic-agent upgrade 8.13.0
   ```
4. In the original window, run `elastic-agent status --output json` again.  Verify that upgrade details are included.

## Related issues


- Relates to https://github.com/elastic/elastic-agent/issues/3119#issuecomment-1743638526
